backend/lib/components/Auth0.js

```diff
@@ -105,10 +106,120 @@ const changePassword = async (token, email) => {
   }
 };
 
+const _getUserDetails = async (token, userId) => {
+  try {
+    const res = await axios.request(
+      `${AUTH_DOMAIN}/api/v2/users/${userId}`,
+      getAuthHeaders(token),
+    );
+    return res.data;
+  } catch (err) {
+    return wrapError(err);
+  }
+};
+
+const _updateUser = async (token, userId, payload) => {
+  try {
+    const res = await axios.request(`${AUTH_DOMAIN}/api/v2/users/${userId}`, {
+      json: payload,
+      ...getAuthHeaders(token),
+    });
+    return res.data;
+  } catch (err) {
+    return wrapError(err);
+  }
+};
+
+const _linkAccounts = async (token, rootUserId, targetUserId) => {
+  const parts = targetUserId.split("|"); // "provider|user_id"
+  const payload = {
+    provider: parts[0],
+    user_id: parts[1],
+  };
+  try {
+    const res = await axios.post(
+      `${AUTH_DOMAIN}/api/v2/users/${rootUserId}/identities`,
+      payload,
+      getAuthHeaders(token),
+    );
+    return res.data;
+  } catch (err) {
+    return wrapError(err);
+  }
+};
+
+const _mergeMetadata = async (token, primaryUserId, secondaryUserId) => {
+  // load both users with metedata.
+  const primaryUser = await _getUserDetails(token, primaryUserId);
+  const secondaryUser = await _getUserDetails(token, secondaryUserId);
+
+  const customizerCallback = function (objectValue, sourceValue) {
+    if (_.isArray(objectValue)) {
+      return sourceValue.concat(objectValue);
+    }
+  };
+  const mergedUserMetadata = _.merge(
+    {},
+    secondaryUser.user_metadata,
+    primaryUser.user_metadata,
+    customizerCallback,
+  );
+  const mergedAppMetadata = _.merge(
+    {},
+    secondaryUser.app_metadata,
+    primaryUser.app_metadata,
+    customizerCallback,
+  );
+  await _updateUser(token, primaryUserId, {
+    user_metadata: mergedUserMetadata,
+    app_metadata: mergedAppMetadata,
+  });
+};
+
+const linkAccounts = async (token, primaryUserId, secondaryUserId) => {
+  try {
+    await _mergeMetadata(token, primaryUserId, secondaryUserId);
+    await _linkAccounts(token, primaryUserId, secondaryUserId);
+  } catch (err) {
+    return wrapError(err);
+  }
+};
+
+const unlinkAccounts = async (token, primaryUserId, secondaryUserId) => {
```

Comment:
```suggestion
```

I don't believe we are planning to add functionality to unlink account so I think we shouldn't include this until that point.

In some rare case where we need to we can do so manually in Auth0 on request.

Reply:
I don't know if our QA team will need such a function to `unlink` or even `delete` accounts, to facilitate the automation test. and I agree it's not what we need for this feature, I will remove it